quartz/plugins/emitters/contentIndex.ts

```diff
@@ -116,50 +118,67 @@ export const ContentIndex: QuartzEmitterPlugin<Partial<Options>> = (opts) => {
-        if (opts?.includeEmptyFiles || (file.data.text && file.data.text !== "")) {
-          linkIndex.set(slug, {
-            title: file.data.frontmatter?.title!,
-            links: file.data.links ?? [],
-            tags: file.data.frontmatter?.tags ?? [],
-            content: file.data.text ?? "",
-            richContent: opts?.rssFullHtml
-              ? escapeHTML(toHtml(tree as Root, { allowDangerousHtml: true }))
-              : undefined,
-            date: date,
-            description: file.data.description ?? "",
-          })
+      const feedIndices: Map<string, ContentIndex> = new Map()
+
+      for (const feed of opts?.feedDirectories!) {
+        const linkIndex: ContentIndex = new Map()
+        for (const [tree, file] of content) {
+          const slug = file.data.slug!
+
+          const date = getDate(ctx.cfg.configuration, file.data) ?? new Date()
+          if (
+            (opts?.includeEmptyFiles || (file.data.text && file.data.text !== "")) &&
+            (slug.startsWith(feed) || feed == "index")
+          ) {
+            linkIndex.set(slug, {
+              title: file.data.frontmatter?.title!,
+              links: file.data.links ?? [],
+              tags: file.data.frontmatter?.tags ?? [],
+              content: file.data.text ?? "",
+              richContent: opts?.rssFullHtml
+                ? escapeHTML(toHtml(tree as Root, { allowDangerousHtml: true }))
+                : undefined,
+              date: date,
+              description: file.data.description ?? "",
+            })
+          }
         }
+        feedIndices.set(feed, linkIndex)
       }
 
+      const siteFeed = feedIndices.get("index")!
       if (opts?.enableSiteMap) {
         emitted.push(
           await write({
             ctx,
-            content: generateSiteMap(cfg, linkIndex),
+            // bfahrenfort: "index" is guaranteed non-null
+            // see directories instantiation and feedIndices.set iterating over directories
+            content: generateSiteMap(cfg, siteFeed),
             slug: "sitemap" as FullSlug,
             ext: ".xml",
           }),
         )
       }
 
       if (opts?.enableRSS) {
-        emitted.push(
-          await write({
-            ctx,
-            content: generateRSSFeed(cfg, linkIndex, opts.rssLimit),
-            slug: "index" as FullSlug,
-            ext: ".xml",
-          }),
-        )
+        var feedPromises: Promise<FilePath>[] = []
+        opts.feedDirectories!.map((feed) => {
+          feedPromises.push(
+            write({
+              ctx,
+              // bfahrenfort: we just generated a feedIndices entry for every directories entry, guaranteed non-null
+              content: generateRSSFeed(cfg, feedIndices.get(feed)!, opts?.rssLimit),
+              slug: feed as FullSlug,
+              ext: ".xml",
+            }),
+          )
+        })
+        emitted.push(...(await Promise.all(feedPromises)))
```

Comment:
so that you don't have to do this.

just need to do

```ts
emitted: Promise<FilePath>[] = []

emitted.push(write...)

...

return Promise.all(emitted)
```

Reply:
Ah, that would require refactoring the whole file since it uses the `emitted.push(await write...)` pattern. That's Jacky's code, so I'd like the ok from them before doing so.